### PR TITLE
[CALCITE-3001] Upgrade to Apache Druid 0.14.0-incubating

### PR DESCRIPTION
**Pull request by nishantmonu51**
Depends on https://github.com/vlsi/calcite-test-dataset/pull/30

### CODE REVIEW

File: druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java [hunk 1]
```diff
@@ -3058,15 +3058,15 @@ private void testCountWithApproxDistinct(boolean approx, String sql, String expe
         + "(count(distinct \"user_id\") * 2) from \"wiki\"";
     wikiApprox(sql)
         .queryContains(druidChecker(druid))
-        .returnsUnordered("EXPR$0=-10590");
+        .returnsUnordered("EXPR$0=100");
```
Comment (jcamachor) [inline druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java]: Is this change of result expected?

File: druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java [hunk 2]
```diff
@@ -3058,15 +3058,15 @@ private void testCountWithApproxDistinct(boolean approx, String sql, String expe
...
 
     // Change COUNT(DISTINCT ...) to APPROX_COUNT_DISTINCT(...) and get
     // same result even if approximation is off by default.
     final String sql2 = "select (approx_count_distinct(\"user_id\") + 100) - "
         + "(approx_count_distinct(\"user_id\") * 2) from \"wiki\"";
     sql(sql2, WIKI)
         .queryContains(druidChecker(druid))
-        .returnsUnordered("EXPR$0=-10590");
+        .returnsUnordered("EXPR$0=100");
```
Comment (jcamachor) [inline druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java]: Is this change of result expected (same as above)?

File: druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java (hunk 1 above)
Comment (nishantmonu51) [reply druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java]: will look into it and verify it.

Comment (nishantmonu51) [reply druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java]: fixed, the cause of failure was missing hyperUnique column in the latest druid default wikipedia ingestion spec, have added the column by copying the ingestion spec file in test vm.

File: druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java (hunk 2 above)
Comment (nishantmonu51) [reply druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java]: fixed.

**Pull request closed without merging**